pkg/image/util/signature/signature.go

```diff
@@ -0,0 +1,65 @@
+package signature
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"time"
+
+	_ "github.com/containers/image/docker"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+// defaultTransportName is the name of default containers/image transport to use
+// to pull signatures.
+const defaultTransportName = "docker"
+
+// defaultGetSignatureTimeout represents a maximum number of seconds that we are
+// willing to wait to get the signature from remote signature store.
+var defaultGetSignatureTimeout = 20 * time.Second
+
+// GetSignatureForImage pulls down image signatures from remote or local signature store.
+// See this document for more details about how to configure the signature stores:
+// https://github.com/containers/image/blob/master/docs/registries.d.md
+func GetSignatureForImage(registryHost, repositoryName, imageName string) ([]imageapi.ImageSignature, error) {
+	t := transports.Get(defaultTransportName)
+	reference, err := t.ParseReference("//" + registryHost + "/" + repositoryName + "@" + imageName)
+	if err != nil {
+		return nil, err
+	}
+	source, err := reference.NewImageSource(&types.SystemContext{}, nil)
```

Comment:
> @mtrmac we are going to use this only to fetch the signature, not fetching the manifest or something. 

I’m afraid that’s not how the implementation works right now, see the `ensureManifestIsLoaded` reference above. Sure, perhaps we should just change that implementation instead of figuring out how to pass the credentials fro OpenShift.

To my knowledge signatures will not be protected at all.